platform-core-module/src/main/java/com/softicar/platform/core/module/email/mailbox/imap/ImapConnection.java

```diff
@@ -57,11 +58,13 @@ public void copyMessageTo(Message message, String targetFolderName) {
 		}
 	}
 
+	@SuppressWarnings("resource")
 	public void moveMessageTo(Message message, String targetFolderName) {
 
 		try {
-			copyMessageTo(message, targetFolderName);
-			message.setFlag(Flags.Flag.DELETED, true);
+			Folder folder = getOpenFolder(message.getFolder().getFullName());
+			IMAPFolder imapFolder = CastUtils.cast(folder);
+			imapFolder.moveMessages(new Message[] { message }, getOpenFolder(targetFolderName));
```

Comment:
would suggest not to use CastUtils
```suggestion
			var imapFolder = (IMAPFolder) getOpenFolder(message.getFolder().getFullName());
			imapFolder.moveMessages(new Message[] { message }, getOpenFolder(targetFolderName));
```

Reply:
ok